python/lsst/ip/isr/calibType.py

```diff
@@ -0,0 +1,630 @@
+                    for rowNum, row in enumerate(data[colName]):
+                        table[rowNum][key] = row
+                else:
+                    table[key] = data[colName]
+
+            metadata = PropertyList()
+            for k, v in data.meta.items():
+                metadata[k] = v
+            table.setMetadata(metadata)
+
+            return cls().fromTable([table])
+
+    def writeText(self, filename):
+        """Write the calibration data to a text file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file to write.
+
+        Returns
+        -------
+        used : `str`
+            The name of the file used to write the data.
+
+        Raises
+        ------
+        RuntimeError :
+            Raised if filename does not end in a known extension
+
+        Notes
+        -----
+        The file is written to YAML/ECSV format and will include any
+        associated metadata.
+        """
+        outDict = self.toDict()
+        if filename.lower().endswith((".yaml")):
+            with open(filename, 'w') as f:
+                yaml.dump(outDict, f)
+        elif filename.lower().endswith((".ecsv")):
+            tableList = self.toTable()
+            tableX = tableList[0]
+
+            table = tableX.asAstropy()
+            table.meta = tableX.getMetadata().toDict()
+
+            # ECSV doesn't support multiple tables per file, so we can only
+            # write the first table.
+            table.write(filename, format="ascii.ecsv")
+        else:
+            raise RuntimeError(f"Attempt to write to a file {filename} "
+                               "that does not end in '.yaml' or '.ecsv'")
+
+        return filename
+
+    @classmethod
+    def readFits(cls, filename):
+        """Read calibration data from a FITS file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Filename to read data from.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.IsrCalib`
+            Calibration contained within the file.
+        """
+        tableList = []
+        tableList.append(afwTable.BaseCatalog.readFits(filename))
+
+        extNum = 2  # Fits indices start at 1, we've read one already.
+        try:
+            newTable = afwTable.BaseCatalog.readFits(filename, extNum)
+            tableList.append(newTable)
+            extNum += 1
+        except Exception:
+            pass
+
+        return cls().fromTable(tableList)
+
+    def writeFits(self, filename):
+        """Write calibration data to a FITS file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Filename to write data to.
+        """
+        tableList = self.toTable()
+
+        tableList[0].writeFits(filename)
+        for table in tableList[1:]:
+            table.writeFits(filename, "a")
+
+    def fromDetector(self, detector):
+        """Set calibration parameters from the detector.
```

Comment:
I find this phrasing confusing.  Maybe "Modify the calibration parameters to match the supplied detector"?